File: ios/Kwift/com/lightningkite/kwift/observables/binding/DateButton.binding.actual.swift
//Stub file made with Kwift 2 (by Lightning Kite)
import Foundation


//--- DateButton.bind(MutableObservableProperty<Date>)
public extension DateButton {
    func bind(_ observable: MutableObservableProperty<Date>) -> Void {
        self.date = observable.value
        observable.addAndRunWeak(referenceA: self) { (this, value) in
            if this.date != value {
                this.date = value
            }
        }
        weak var observableWeak = observable
        self.onDateEntered.addWeak(self) { (self, value) in
            if observableWeak?.value != value {
                observableWeak?.value = value
            }
        }
    }
    func bind(date: MutableObservableProperty<Date>) -> Void {
        return bind(date)
    }
}

//--- TimeButton.bind(MutableObservableProperty<Date>, Int)
public extension TimeButton {
    func bind(_ observable: MutableObservableProperty<Date>, _ minuteInterval: Int32 = 1) -> Void {
        self.minuteInterval = Int(minuteInterval)
        self.date = observable.value
        observable.addAndRunWeak(referenceA: self) { (this, value) in
            if this.date != value {
                this.date = value
            }
        }
        weak var observableWeak = observable
        self.onDateEntered.addWeak(self) { (self, value) in
            if observableWeak?.value != value {
                observableWeak?.value = value
            }
        }
    }
    func bind(date: MutableObservableProperty<Date>, minuteInterval: Int32 = 1) -> Void {
        return bind(date, minuteInterval)
    }
}

//--- DateButton.bindDateAlone(MutableObservableProperty<DateAlone>)
public extension DateButton {
    func bindDateAlone(_ observable: MutableObservableProperty<DateAlone>) -> Void {
        observable.addAndRunWeak(referenceA: self) { (this, value) in
            if this.date.dateAlone != value {
                this.date = dateFrom(value, Date().timeAlone)
            }
        }
        weak var observableWeak = observable
        self.onDateEntered.addWeak(self) { (self, value) in
            let newValue = self.date.dateAlone
            if observableWeak?.value != newValue {
                observableWeak?.value = newValue
            }
        }
    }
    func bindDateAlone(date: MutableObservableProperty<DateAlone>) -> Void {
        return bindDateAlone(date)
    }
}

//--- TimeButton.bindTimeAlone(MutableObservableProperty<TimeAlone>, Int)
public extension TimeButton {
    func bindTimeAlone(_ observable: MutableObservableProperty<TimeAlone>, _ minuteInterval: Int32 = 1) -> Void {
        self.minuteInterval = Int(minuteInterval)
        observable.addAndRunWeak(referenceA: self) { (this, value) in
            if this.date.timeAlone != value {
                this.date = dateFrom(Date().dateAlone, value)
            }
        }
        weak var observableWeak = observable
        self.onDateEntered.addWeak(self) { (self, value) in
            let newValue = self.date.timeAlone
            if observableWeak?.value != newValue {
                observableWeak?.value = newValue
            }
        }
    }
    func bindTimeAlone(date: MutableObservableProperty<TimeAlone>, minuteInterval: Int32 = 1) -> Void {
        return bindTimeAlone(date, minuteInterval)
    }
}

File: android/src/main/java/com/lightningkite/kwift/observables/binding/DateButton.binding.actual.kt
package com.lightningkite.kwift.observables.binding

import com.lightningkite.kwift.observables.*
import com.lightningkite.kwift.rx.addWeak
import com.lightningkite.kwift.time.*
import com.lightningkite.kwift.views.android.DateButton
import com.lightningkite.kwift.views.android.TimeButton
import java.util.*


fun DateButton.bind(date: MutableObservableProperty<Date>) {
    date.observableNN.addWeak(this) { self, it ->
        self.date = it
    }
    this.onDateEntered.addWeak(this) { self, it ->
        date.value = it
    }
}

fun TimeButton.bind(date: MutableObservableProperty<Date>, minuteInterval: Int = 1) {
    this.minuteInterval = minuteInterval
    date.observableNN.addWeak(this) { self, it ->
        self.date = it
    }
    this.onDateEntered.addWeak(this) { self, it ->
        date.value = it
    }
}

fun DateButton.bindDateAlone(date: MutableObservableProperty<DateAlone>) {
    date.observableNN.addWeak(this, { self, it ->
        self.date = dateFrom(it, Date().timeAlone)
    })
    this.onDateEntered.addWeak(this) { self, it ->
        date.value = it.dateAlone
    }
}

fun TimeButton.bindTimeAlone(date: MutableObservableProperty<TimeAlone>, minuteInterval: Int = 1) {
    this.minuteInterval = minuteInterval
    date.observableNN.addWeak(this) { self, it ->
        self.date = dateFrom(Date().dateAlone, it)
    }
    this.onDateEntered.addWeak(this) { self, it ->
        date.value = it.timeAlone
    }
}
